src/widget/wspinny.cpp

```diff
@@ -16,6 +16,102 @@
+}
+
+void WSpinnyLayout::addItem(QLayoutItem* item) {
+    m_pSpinnyItem = item;
+    m_pSpinnyWidget = item->widget();
+}
+
+QSize WSpinnyLayout::sizeHint() const {
+    qDebug() << "WSpinnyLayout::sizeHint. Geometry:" << geometry();
+    if (m_pSpinnyWidget == nullptr) {
+        return QSize(0,0);
+    }
+    return QSize(parentWidget()->height(), parentWidget()->height());
+    if (geometry().width() > geometry().height()) {
+        return QSize(geometry().height(), geometry().height());
+    }
+    return QSize(geometry().width(), geometry().width());
+//     if (m_pSpinnyWidget->width() > m_pSpinnyWidget->height()) {
+//         return QSize(m_pSpinnyWidget->height(), m_pSpinnyWidget->height());
+//     }
+//     return QSize(m_pSpinnyWidget->width(), m_pSpinnyWidget->width());
+}
+
+QSize WSpinnyLayout::minimumSize() const {
+    qDebug() << "WSpinnyLayout::maximumSize. Geometry:" << geometry();
+    if (m_pSpinnyWidget == nullptr) {
+        return QSize(0,0);
+    }
+    return QSize(parentWidget()->height(), parentWidget()->height());
+    if (geometry().width() > geometry().height()) {
+        return QSize(geometry().height(), geometry().height());
+    }
+    return QSize(geometry().width(), geometry().width());
+//     if (m_pSpinnyWidget->width() > m_pSpinnyWidget->height()) {
+//         return QSize(m_pSpinnyWidget->height(), m_pSpinnyWidget->height());
+//     }
+//     return QSize(m_pSpinnyWidget->width(), m_pSpinnyWidget->width());
+}
+
+QSize WSpinnyLayout::maximumSize() const {
+    qDebug() << "WSpinnyLayout::maximumSize. Geometry:" << geometry();
+    if (m_pSpinnyWidget == nullptr) {
+        return QSize(0,0);
+    }
+    return QSize(parentWidget()->height(), parentWidget()->height());
+    if (geometry().width() > geometry().height()) {
+        return QSize(geometry().height(), geometry().height());
+    }
+    return QSize(geometry().width(), geometry().width());
+//     if (m_pSpinnyWidget->width() > m_pSpinnyWidget->height()) {
+//         return QSize(m_pSpinnyWidget->height(), m_pSpinnyWidget->height());
+//     }
+//     return QSize(m_pSpinnyWidget->width(), m_pSpinnyWidget->width());
+}
+
+QLayoutItem *WSpinnyLayout::itemAt(int i) const {
+    Q_UNUSED(i);
+    if (m_pSpinnyItem == nullptr) {
+        return 0;
+    }
+    return m_pSpinnyItem;
+}
+
+QLayoutItem *WSpinnyLayout::takeAt(int i) {
+    Q_UNUSED(i);
+    if (m_pSpinnyItem == nullptr) {
+        return 0;
+    }
+    return m_pSpinnyItem;
+}
+
+int WSpinnyLayout::count() const {
+    if (m_pSpinnyItem == nullptr) {
+        return 0;
+    }
+    return 1;
+}
+
+void WSpinnyLayout::setGeometry(const QRect &rect) {
+    qDebug() << "WSpinnyLayout::setGeometry" << rect;
+    if (m_pSpinnyItem == nullptr || m_pSpinnyWidget == nullptr) {
+        return;
+    }
+
+    if (rect.height() > rect.width()) {
+        m_pSpinnyItem->setGeometry(QRect(0, 0, rect.width(), rect.width()));
+        QLayout::setGeometry(QRect(0, 0, rect.width(), rect.width()));
```

Comment:
I haven't found any documentation on what calling QLayout::setGeometry() in a QLayout subclass' setGeometry method does, but I see it in all of Qt's examples.